pkg/apis/pipeline/v1/matrix_types_test.go

```diff
@@ -915,11 +915,20 @@ func TestPipelineTask_CountCombinations(t *testing.T) {
 				}},
 			}},
 		want: 7,
+	}, {
+		name: "matrix params with string value containing variable reference",
+		matrix: &v1.Matrix{
+			Params: v1.Params{{
+				Name: "GOARCH", Value: v1.ParamValue{ArrayVal: []string{"linux/amd64", "linux/ppc64le", "linux/s390x"}},
+			}, {
+				Name: "version", Value: v1.ParamValue{StringVal: "$(tasks.platforms.results.str[*])"}},
```

Comment:
The problem to be solved this time involves referencing an array parameter within a string value.